File: test-suite/src/test/java/io/micronaut/docs/http/server/secondary/SecondaryNettyServer.java
package io.micronaut.docs.http.server.secondary;

// tag::imports[]
import java.util.Collections;
import io.micronaut.context.annotation.Bean;
import io.micronaut.context.annotation.Context;
import io.micronaut.context.annotation.Factory;
import io.micronaut.context.annotation.Requires;
import io.micronaut.context.env.Environment;
import io.micronaut.core.util.StringUtils;
import io.micronaut.discovery.ServiceInstanceList;
import io.micronaut.discovery.StaticServiceInstanceList;
import io.micronaut.http.server.netty.NettyEmbeddedServer;
import io.micronaut.http.server.netty.NettyEmbeddedServerFactory;
import io.micronaut.http.server.netty.configuration.NettyHttpServerConfiguration;
import io.micronaut.http.ssl.ServerSslConfiguration;
import jakarta.inject.Named;
// end::imports[]

@Requires(property = "secondary.enabled", value = StringUtils.TRUE)
// tag::class[]
@Factory
public class SecondaryNettyServer {
    public static final String SERVER_ID = "another"; // <1>

    @Named(SERVER_ID)
    @Context
    @Bean(preDestroy = "close") // <2>
    @Requires(beans = Environment.class)
    NettyEmbeddedServer nettyEmbeddedServer(NettyEmbeddedServerFactory serverFactory) { // <3>
        // configure server programmatically
        final NettyHttpServerConfiguration configuration =
                new NettyHttpServerConfiguration(); // <4>
        final ServerSslConfiguration sslConfiguration = new ServerSslConfiguration(); // <5>
        sslConfiguration.setBuildSelfSigned(true);
        sslConfiguration.setEnabled(true);
        sslConfiguration.setPort(-1); // random port
        final NettyEmbeddedServer embeddedServer = serverFactory.build(configuration, sslConfiguration); // <6>
        embeddedServer.start(); // <7>
        return embeddedServer; // <8>
    }

    @Bean
    ServiceInstanceList serviceInstanceList( // <9>
            @Named(SERVER_ID) NettyEmbeddedServer nettyEmbeddedServer) {
        return new StaticServiceInstanceList(
                SERVER_ID,
                Collections.singleton(nettyEmbeddedServer.getURI())
        );
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/http/server/secondary/SecondaryNettyServer.py
from typing import Annotated

import java

# tag::imports[]
from jakarta.inject import Named
from micronaut.context.annotation import Bean, Context, Factory, Requires
from micronaut.context.env import Environment
from micronaut.core.util import StringUtils
from micronaut.discovery import ServiceInstanceList, StaticServiceInstanceList
from micronaut.http.server.netty import NettyEmbeddedServer, NettyEmbeddedServerFactory
from micronaut.http.server.netty.configuration import NettyHttpServerConfiguration
from micronaut.http.ssl import ServerSslConfiguration
# end::imports[]

Collections = java.type("java.util.Collections")


@Requires(property="secondary.enabled", value=StringUtils.TRUE)
# tag::class[]
@Factory
class SecondaryNettyServer:
    SERVER_ID = "another"  # <1>

    @Named("another")
    @Context
    @Bean(preDestroy="close")  # <2>
    @Requires(beans=Environment)
    def nettyEmbeddedServer(
        self, serverFactory: NettyEmbeddedServerFactory
    ) -> NettyEmbeddedServer:  # <3>
        configuration = NettyHttpServerConfiguration()  # <4>
        sslConfiguration = ServerSslConfiguration()  # <5>
        sslConfiguration.setBuildSelfSigned(True)
        sslConfiguration.setEnabled(True)
        sslConfiguration.setPort(-1)  # random port
        embeddedServer = serverFactory.build(configuration, sslConfiguration)  # <6>
        embeddedServer.start()  # <7>
        return embeddedServer  # <8>

    @Bean
    def serviceInstanceList(  # <9>
        self,
        nettyEmbeddedServer: Annotated[NettyEmbeddedServer, Named("another")],
    ) -> ServiceInstanceList:
        return StaticServiceInstanceList(
            self.SERVER_ID,
            Collections.singleton(nettyEmbeddedServer.getURI()),
        )
# end::class[]
